src/sentry/lang/native/plugin.py

```diff
@@ -443,5 +438,9 @@ def report_error(e):
 class NativePlugin(Plugin2):
     can_disable = False
 
-    def get_event_preprocessors(self, **kwargs):
-        return [preprocess_apple_crash_event, resolve_frame_symbols]
+    def get_event_preprocessors(self, data, **kwargs):
+        if data.get('sentry.interfaces.AppleCrashReport'):
```

Comment:
IMO generators are problematic but it's better to return a single data type. Peopl too frequently abuse pythons lack of typing but doing extremely slow isinstance checks. I don't care what it returns but it should always be the same. Either way all other similar hooks are returning lists today.